cpg_workflows/workflow.py

```diff
@@ -679,9 +723,17 @@ def _is_reusable(self, expected_out: ExpectedResultT) -> tuple[bool, Path | None
 
         if get_config()['workflow'].get('check_expected_outputs'):
             paths = path_walk(expected_out, set())
-            first_missing_path = next((p for p in paths if not exists(p)), None)
             if not paths:
                 return False, None
+
+            # check against the pre-scanned files if possible
+            if existing_outputs:
+
+                first_missing_path = exists_on_pre_collected(paths, existing_outputs)
```

Comment:
I think you need to workshop this function name a bit more. The `next(… if not exists(p), …)` previous code gives us the clue: its functionality is really closer to `not_exists_on_pre_collected()`.

Reply:
Fair point, I've replaced it with `missing_from_pre_collected` as both None and "a file" semantically fit with that name